x-pack/plugins/rule_registry/server/alert_data_client/tests/get.test.ts

```diff
@@ -0,0 +1,160 @@
+  esClient: esClientMock,
+  auditLogger,
+};
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('get()', () => {
+  test('calls ES client with given params', async () => {
+    const alertsClient = new AlertsClient(alertsClientParams);
+    esClientMock.get.mockResolvedValueOnce(
+      elasticsearchClientMock.createApiResponse({
+        body: {
+          found: true,
+          _type: 'alert',
+          _index: '.alerts-observability-apm',
+          _id: 'NoxgpHkBqbdrfX07MqXV',
+          _source: {
+            'rule.id': 'apm.error_rate',
+            message: 'hello world 1',
+            'kibana.rac.alert.owner': 'apm',
+            'kibana.rac.alert.status': 'open',
+          },
+        },
+      })
+    );
+    const result = await alertsClient.get({ id: '1', indexName: '.alerts-observability-apm' });
+    expect(result).toMatchInlineSnapshot(`
+      Object {
+        "kibana.rac.alert.owner": "apm",
+        "kibana.rac.alert.status": "open",
+        "message": "hello world 1",
+        "rule.id": "apm.error_rate",
+      }
+    `);
+    expect(esClientMock.get).toHaveBeenCalledTimes(1);
+    expect(esClientMock.get.mock.calls[0]).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "id": "1",
+          "index": ".alerts-observability-apm",
+        },
+      ]
+    `);
+    expect(auditLogger.log).toHaveBeenCalledWith({
+      error: undefined,
+      event: { action: 'alert_get', category: ['database'], outcome: 'success', type: ['access'] },
+      message: 'User has accessed alert [id=1]',
+    });
+  });
+
+  test(`throws an error if ES client get fails`, async () => {
+    const error = new Error('something when wrong');
+    const alertsClient = new AlertsClient(alertsClientParams);
+    esClientMock.get.mockRejectedValue(error);
+
+    await expect(
+      alertsClient.get({ id: '1', indexName: '.alerts-observability-apm' })
+    ).rejects.toThrowErrorMatchingInlineSnapshot(`"something when wrong"`);
+    expect(auditLogger.log).toHaveBeenCalledWith({
+      error: { code: 'Error', message: 'something when wrong' },
+      event: { action: 'alert_get', category: ['database'], outcome: 'failure', type: ['access'] },
+      message: 'Failed attempt to access alert [id=1]',
+    });
+  });
+
+  describe('authorization', () => {
+    beforeEach(() => {
+      esClientMock.get.mockResolvedValueOnce(
+        elasticsearchClientMock.createApiResponse({
+          body: {
+            found: true,
+            _type: 'alert',
+            _index: '.alerts-observability-apm',
+            _id: 'NoxgpHkBqbdrfX07MqXV',
+            _source: {
+              'rule.id': 'apm.error_rate',
+              message: 'hello world 1',
+              'kibana.rac.alert.owner': 'apm',
+              'kibana.rac.alert.status': 'open',
+            },
+          },
+        })
+      );
+    });
+
+    test('returns alert if user is authorized to read alert under the consumer', async () => {
```

Comment:
If I find myself typing "if" or "when" in an `it` block I usually break it up into something like:

```
describe('when the user is authorized to read the alert under the consumer', () => {
  it('returns the alert', () => ...
```